src/constants.js

```diff
@@ -1,75 +1,65 @@
 import * as styledSystem from 'styled-system'
+import { compose } from 'styled-system'
+import systemPropTypes from '@styled-system/prop-types'
+import themeGet from '@styled-system/theme-get'
 import theme from './theme'
 
-const {compose, get: getKey, themeGet} = styledSystem
+const { get: getKey } = styledSystem
 
 export const get = key => themeGet(key, getKey(theme, key))
 
-export const COMMON_LIST = ['color', 'space']
-
-export const BORDER_LIST = ['borders', 'borderColor', 'boxShadow', 'borderRadius']
-
-export const TYPOGRAPHY_LIST = [
-  // typography props
-  'fontFamily',
-  'fontSize',
-  'fontStyle',
-  'fontWeight',
-  'lineHeight',
-  'textAlign'
-]
+export const COMMON = compose(
+  styledSystem.space,
+  styledSystem.color,
+)
+COMMON.propTypes = {
+  ...systemPropTypes.space,
+  ...systemPropTypes.color,
+}
+export const BORDER = compose(
+  styledSystem.border,
+  styledSystem.shadow,
+)
+BORDER.propTypes = {
+  ...systemPropTypes.border,
+  ...systemPropTypes.shadow,
+}
 
-export const LAYOUT_LIST = [
-  // layout props
-  'display',
-  'size',
-  'width',
-  'height',
-  'minWidth',
-  'minHeight',
-  'maxWidth',
-  'maxHeight',
-  'overflow',
-  'verticalAlign'
-]
+// these are 1:1 with styled-system's API now,
+// so you could consider dropping the abstraction
```

Comment:
🎉 I'm super open to importing `typography`, `layout`, etc. directly from `styled-system`